pulp/apis/cplex_api.py

```diff
@@ -430,6 +444,47 @@ def cplex_var_types(var):
                 self.solverModel.MIP_starts.add(
                     cplex.SparsePair(ind=ind, val=val), effort, "1"
                 )
+            for param, value in self.solverParams.items():
+                self.set_param(param, value)
+
+        def set_param(self, name: str, value):
+            """
+            Sets a parameter value using its name.
+            """
+            param = self.search_param(name=name)
+            param.set(value)
+
+        def get_param(self, name: str):
+            """
+            Returns the value of a named parameter by searching within the instance's parameters.
+            """
+            param = self.search_param(name=name)
+            return param.get()
+
+        def search_param(self, name: str):
+            """
+            Searches for a solver model parameter by its name and returns the corresponding attribute.
+
+            The method takes a parameter name string, processes it to remove the "parameters." prefix
+            and splits it by periods to traverse the attribute hierarchy of the solver model's parameters.
+            """
+            name = name.replace("parameters.", "")
+            param = self.solverModel.parameters
```

Comment:
`self.solverModel` may not exist if `self.buildSolverModel` has not been called yet. Maybe we need to initialized it with None at __init__ and then check if it's none before using it